pipeline/data/dataset_importer.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+"""
+Downloads a dataset and runs augmentation if needed
+
+Example:
+    SRC=ru TRG=en python pipeline/data/dataset_importer.py \
+        --type=corpus \
+        --dataset=sacrebleu_aug-mix_wmt19 \
+        --output_prefix=$(pwd)/test_data/augtest
+"""
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+from typing import Iterable, List
+
+from opustrainer.modifiers.surface import TitleCaseModifier, UpperCaseModifier
+from opustrainer.modifiers.typos import TypoModifier
+from opustrainer.types import Modifier
+
+
+class CompositeModifier:
+    def __init__(self, modifiers: List[Modifier]):
+        self._modifiers = modifiers
+
+    def __call__(self, batch: List[str]) -> Iterable[str]:
+        for mod in self._modifiers:
+            batch = list(mod(batch))
+
+        return batch
+
+
+MIX_PROB = 0.1  # 10% will be augmented in the mix
+
+modifier_map = {
+    "aug-typos": TypoModifier(MIX_PROB),
+    "aug-title": TitleCaseModifier(MIX_PROB),
+    "aug-title-strict": TitleCaseModifier(1.0),
+    "aug-upper": UpperCaseModifier(MIX_PROB),
+    "aug-upper-strict": UpperCaseModifier(1.0),
+    "aug-mix": CompositeModifier(
+        [
+            # TODO: enable typos, issue https://github.com/hplt-project/OpusTrainer/issues/40
+            # TypoModifier(NOISE_RATE),
+            TitleCaseModifier(MIX_PROB),
+            UpperCaseModifier(MIX_PROB),
+        ]
+    ),
+}
+
+
+def run_cmd(cmd: List[str]):
+    result = None
+    try:
+        result = subprocess.run(
+            cmd,
+            universal_newlines=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=False,
+        )
+        result.check_returncode()
+    except:
+        if result:
+            print(result.stdout)
+        raise
+
+    print(result.stdout)
+
+
+# we plan to use it only for small evaluation datasets
+def augment(output_prefix: str, aug_modifer: str):
```

Comment:
This augment function is pretty long and handles multiple things (file handling, augmentation, compression). It might be good to split it into smaller functions.